### ISSUE TITLE
Create New CRS elements

### ISSUE THREAD

**Issue by caprenter** [IATI/IATI-Schemas#4]
Start at line 311 of this:
https://docs.google.com/spreadsheet/ccc?key=0AnWngmdQt3stdDcwRHJxaXgtLWVoY2N5VWt4ck1oeGc#gid=1

**Comment by davidmegginson** [comment]
Added crs-add/aidtype-flag

Still to do: loan-terms, loan-status, and fss

**Comment by caprenter** [comment]
loan-terms has new codelists associated with it: https://docs.google.com/spreadsheet/ccc?key=0AnWngmdQt3stdDF3SDcwUEQwWnBuVE9IWS11bFVleWc#gid=0

There is an issue with the where these are to be attached to the element - in the codelists spec it looks like they are constraining the values of the text used within the element, rather than being added via an attribute as is usual with 'codelist' values. You may need to just check that the XML caters for the new codes.

This was flagged by Kit Wallace who is working on getting the 1.03 codelists together. https://github.com/KitWallace/AIDVIEW-DB/issues/120

**Comment by davidmegginson** [comment]
Agreed - loan-terms needs some cleanup.  Here are my recommendations:
- add loan-terms/commit-date/@iso-date to hold the ISO date, to align with other date elements
- add loan-terms/repayment-type/@code, to align with other code elements
- rename loan-terms/nopa to loan-terms/repayment-plan for clarity
- add loan-terms/repayment-plan/@code to align with other code elements
- change loan-terms/rate-1 element to loan-terms/@variable-rate attribute
- change loan-terms/rate-2 element to loan-terms/@fixed-rate attribute
- rename loan-terms/repayment-first to loan-terms/repayment-start-date for clarity
- add loan-terms/repayment-start-date/@iso-date, to align with other date elements
- rename loan-terms/repayment-last to loan-terms/repayment-end-date for clarify
- add loan-terms/repayment-end-date/@iso-date, to align with other date elements

In other words, it would look like this:

```
<loan-terms variable-rate="4%" fixed-rate="3%">
  <repayment-type code="1">Equal Principal Payments (EPP)</repayment-type>
  <repayment-plan code="4">Quarterly</repayment-plan>
  <repayment-start-date iso-date="2014-01-01"/>
  <repayment-end-date iso-date="2020-12-31"/>
</loan-terms>
```

Where would be the right forum to bring these up?  Should we just talk to Bill, or consult more widely?

**Comment by bill-anderson** [comment]
Structure is good for me but not name changes:

These elements will only be used by CRS reporters who are used to these names as CRS++ column definitions

If no variable rate is present rate-1 will be the fixed rate. Changing these names adds extra processing logic into IATI and from IATI to CRS++.

Start date and end date do not have the same meaning as first and final, so I would leave as is

Therefore, I suggest ...

```
<loan-terms rate-1="4%" rate-2="3%">
  <repayment-type code="1">Equal Principal Payments (EPP)</repayment-type>
  <repayment-plan code="4">Quarterly</repayment-plan>
  <repayment-first-date iso-date="2014-01-01"/>
  <repayment-final-date iso-date="2020-12-31"/>
</loan-terms>
```

**Comment by caprenter** [comment]
Edited Bill's comment to show the markup. N.B. Indent code by four spaces for the issue list to recognise it as such.

**Comment by davidmegginson** [comment]
working on loan-terms now ...

**Comment by davidmegginson** [comment]
Implemented loan-terms, following @bill-anderson 's modifications.  Here's an example:

```
<loan-terms rate-1="4" rate-2="3">
  <repayment-type code="1">Equal Principal Payments (EPP)</repayment-type>
  <repayment-plan code="4">Quarterly</repayment-plan>
  <commitment-date iso-date="2013-09-01"/>
  <repayment-first-date iso-date="2014-01-01"/>
  <repayment-final-date iso-date="2020-12-31"/>
</loan-terms>
```

Pushed changes upstream in the 1.03dev git branch. Still need to do load-status and fss.

**Comment by davidmegginson** [comment]
Here's my planned markup for loan-status (no changes to what's in the spreadsheet):

```
<loan-status year="2014" currency="GBP" value-date="2013-05-24">
  <interest-received>200000</interest-received>
  <principal-outstanding>1500000</principal-outstanding>
  <principal-arrears>0</principal-arrears>
  <interest-arrears>0</interest-arrears>
  <principal-first-year>500000</principal-first-year>
  <interest-first-year>20000</interest-first-year>
</loan-status>
```

**Comment by davidmegginson** [comment]
Implemented crs-add/loan-status and pushed upstream to 1.03dev.  Only crs-add/fss remaining.

**Comment by davidmegginson** [comment]
Here's my planned markup for crs-add/fss:

```
<fss extraction-date="2013-05-31" priority="true" phaseout-year="2016">
  <forecast year="2014" currency="GBP">10000</forecast>
  <forecast year="2015" currency="GBP">20000</forecast>
</fss>
```

Note that I've left out forecast/@value-date for now, since it's not in the standard, but I've created issue #13 to check whether we need it.  I've also renamed fss/@phaseout to fss/@phaseout-year, for clarity.

**Comment by davidmegginson** [comment]
Added fss.  Note that the standard does not state that fss appears inside crs-add -- please confirm.

@caprenter - can you please review the schema and confirm that this task is complete (all the crs-add markup is added)?  See tests/activity-tests/should-pass/31-crs-add.xml for the test that covers this new markup.

**Comment by bill-anderson** [comment]
fss should not be inside crs-add

Creditor Reporting System and Forward Spending Survey are two separate OECD DAC products

**Comment by caprenter** [comment]
I've committed a change that should fix. Please review.

**Comment by bill-anderson** [comment]
Sorry guys, two elements need removing as per Herman's note - last comment in http://support.iatistandard.org/entries/22874883-Extend-the-IATI-standard-with-elements-to-cover-CRS- 

loan-status/principal-first-year
loan-status/interest-first-year

crs-add will only be used for future crs++ reporting, not backward looking, so these two can be dropped.

**Comment by davidmegginson** [comment]
Assigned to @caprenter for feedback and verification.  All tests pass under tests/activity-tests/

**Comment by caprenter** [comment]
Looks good.
